### PR TITLE
Add source and origin to posts

### PR DESCRIPTION
**Pull request by tylerelston**
Added source and origin fields to posts for reblogging

Origin: original post link
Source: link you got the post from

Both origin and source are set to the post's ID link when first created

image

### CODE REVIEW

**Review by debecerra — APPROVED** [review]
I think you're missing the migrations file.

Also could consider just returning the self.get_id in the origin and source field when getting the JSON of a local post, instead of saving origin and source fields. Don't really need to store the same thing three times and wouldn't need new migrations in this case. But this is very minor and optional.

Just the migrations issue and it's good, thanks

**Comment by tylerelston** [comment]
> I think you're missing the migrations file.
> 
> Also could consider just returning the self.get_id in the origin and source field when getting the JSON of a local post, instead of saving origin and source fields. Don't really need to store the same thing three times and wouldn't need new migrations in this case. But this is very minor and optional.
> 
> Just the migrations issue and it's good, thanks

I don't think we would return self.get_id for source/origin inside as_json since a re-shared post could have a different source/origin while a brand new post would have the same link for source/origin

**Comment by debecerra** [comment]
> 

Oh, you're definitely right, I forgot how the sharing works.

File: code/socialDistribution/tests.py [hunk 1]
```diff
@@ -85,7 +85,24 @@ def test_share_public_post(self):
...
+        visibility = LocalPost.Visibility.PUBLIC
+        post = PostBuilder().visibility(visibility).build()
+        
+        self.client.post('socialDistribution:share-post', id=post.id)
+        middlePost = LocalPost.objects.latest("published")
+        self.client.post('socialDistribution:share-post', id=middlePost.id)
+        latestPost = LocalPost.objects.latest("published")
+        
+        self.assertEquals(latestPost.visibility, LocalPost.Visibility.PUBLIC)
+        self.assertEqual(latestPost.origin, post.get_id())
```
Comment (JawadHossain) [inline code/socialDistribution/tests.py:104]: 💯

**Review by JawadHossain — APPROVED** [review]
Nice!

**Pull request merged**